staging/src/k8s.io/client-go/dynamic/fake/simple.go

```diff
@@ -116,10 +121,23 @@ func NewSimpleDynamicClientWithCustomListKinds(scheme *runtime.Scheme, gvrToList
 	cs.AddWatchReactor("*", func(action testing.Action) (handled bool, ret watch.Interface, err error) {
 		gvr := action.GetResource()
 		ns := action.GetNamespace()
+
+		if replable, ok := o.(ReplayableWatch); ok {
+			if watchAction, ok := action.(testing.WatchAction); ok && watchAction.GetWatchRestrictions().RequiresInitialEvents {
+				klog.Infof("replaying initial events")
+				watch, err := replable.WatchWithReplay(gvr, ns)
```

Comment:
I had inclination to add another argument to `Watch`. It is part of an interface. I think it should be a part of it, or at least an options struct argument (Would cause ripple for every usage)

opintions?